cpp/src/shap/permutation_shap.cu

```diff
@@ -0,0 +1,187 @@
+/*
+ * Copyright (c) 2020, NVIDIA CORPORATION.
```

Comment:
That's an excellent question. In general, our `include` folder follows the scikit-learn folder structure, while our `src` folder is more fine grained and divided in folder per algo (for the most part). This makes sense, since some algos have a significant number of files, so the scikit structure would lead to some _very_ overweight folders or a more nested `src` folder, neither which is ideal. So following this, I think having a `explainer` group in the scikit learn structure `include` (and `python/cuml`) is consistent there, while here grouping the `shap` code from future explainer code seems to be the better choices. What do you think?